pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java

```diff
@@ -486,69 +447,44 @@ private void setGroupByResults(BrokerResponseNative brokerResponseNative, boolea
       finalResultMaps[i] = finalResultMap;
     }
 
-    int aggregationNumsInFinalResult = 0;
-    for (int i = 0; i < _numAggregationFunctions; i++) {
-      if (aggregationFunctionsSelectStatus[i]) {
-        aggregationNumsInFinalResult++;
-      }
-    }
-
-    if (aggregationNumsInFinalResult > 0) {
```

Comment:
Looks like this piece of code was redundant and that'w why you invoke trimming on already compute finalResultMaps in the new code?

Reply:
The `aggregationFunctionsSelectStatus` is redundant (was introduced for HAVING clause), and thus this if check is no longer valid (always true). This part of the code is the same as before, the only change is the indentation.